lib/remove.js

```diff
@@ -19,23 +19,19 @@ function remove (gyp, argv, callback) {
   log.verbose('remove', 'using node-gyp dir:', devDir)
 
   // get the user-specified version to remove
-  var v = argv[0] || gyp.opts.target
-  log.verbose('remove', 'removing target version:', v)
+  var version = argv[0] || gyp.opts.target
+  log.verbose('remove', 'removing target version:', version)
 
-  if (!v) {
+  if (!version) {
     return callback(new Error('You must specify a version number to remove. Ex: "' + process.version + '"'))
   }
 
-  // parse the version to normalize and make sure it's valid
-  var version = semver.parse(v)
-
-  if (!version) {
-    return callback(new Error('Invalid version number: ' + v))
+  var versionSemver = semver.parse(version)
+  if (versionSemver) {
+    // flatten the version Array into a String
+    version = versionSemver.version
   }
 
-  // flatten the version Array into a String
-  version = version.version
-
   var versionPath = path.resolve(gyp.devDir, version)
```

Comment:
that comes in through `versionDir` from install, and if you want to remove an io.js version then you'd need to call it as `io.js-3.3.0`

I doubt very much whether this is ever called directly though so I'm not so concerned about tuning this to some imaginary scenario of how someone might want to use it